### ISSUE TITLE
cvat sdk (low and high level apis) loses tasks

### ISSUE THREAD

**Issue by dmg-ai** [opencv/cvat#6876]
Library version: `cvat_sdk==2.3.0`

Used High-level API code to retrieve list of project tasks:
```
from cvat_sdk import make_client
client = make_client(
    host=cvat_host,
    credentials=cvat_creds,
)
project = client.projects.retrieve(project_num)
tasks_list = project.get_tasks()
```
In this case, i got 10 tasks of 16. But the most interesting thing is that on the project page with a list of tasks in cvat itself, the tasks are divided into 2 pages and the first contains 10 tasks, and the second 6. Accordingly, I just get a list from the first page of 10 tasks.

The same happens with Low-level API - I get 10 tasks of 16:
```
from cvat_sdk.api_client import Configuration, ApiClient, models, apis, exceptions

configuration = Configuration(
    host=cvat_host,
    username=cvat_username
    password=cvat_password,
)
api_client = ApiClient(configuration)
print(len(api_client.projects_api.list_tasks(project_num)[0]["results"]))
```
The last raw prints the length of results dictionary that contains tasks.

**Comment by zhiltsov-max** [comment]
Hi, this problem was fixed in https://github.com/opencv/cvat/pull/5557 (v2.4.0+). Prior this change, only the first page (up to 10 entries) had been requested in this function. If you want to retrieve all the list, please use the low-level API and the `get_paginated_collection()` function:

```python
from cvat_sdk.core.helpers import get_paginated_collection
from cvat_sdk.core.proxies.tasks import Task
from cvat_sdk import make_client

format_name = "CVAT for images 1.1"

with make_client("localhost", port=8080, credentials=("admin", "pass")) as client:
    tasks = [
        Task(client, m)
        for m in get_paginated_collection(
            client.api_client.projects_api.list_tasks_endpoint, id=project_id
        )
    ]

    for task in tasks:
        task.export_dataset(format_name, f"{task.id}-{task.name}-{format_name}.zip")
        
```

**Comment by dmg-ai** [comment]
> Hi, this problem was fixed in #5557 (v2.4.0+). Prior this change, only the first page (up to 10 entries) had been requested in this function. If you want to retrieve all the list, please use the low-level API and the `get_paginated_collection()` function:
> 
> ```python
> from cvat_sdk.core.helpers import get_paginated_collection
> from cvat_sdk.core.proxies.tasks import Task
> from cvat_sdk import make_client
> 
> format_name = "CVAT for images 1.1"
> 
> with make_client("localhost", port=8080, credentials=("admin", "pass")) as client:
>     tasks = [
>         Task(client, m)
>         for m in get_paginated_collection(
>             client.api_client.projects_api.list_tasks_endpoint, id=project_id
>         )
>     ]
> 
>     for task in tasks:
>         task.export_dataset(format_name, f"{task.id}-{task.name}-{format_name}.zip")
>         
> ```

Hi! I tried the way you suggested but for me it doesn't work. I got the error below:

<img width="924" alt="image" src="https://github.com/opencv/cvat/assets/55949105/20035228-2fba-4c5d-8b03-e6b89f129ce2">
 
When I create client object:
```
 client = make_client(host=host, credentials=("admin", "pass"))
```

I get the warning message:

`Server version '2.3' is not compatible with SDK version '2.7.2'. Some SDK functions may not work properly with this server. You can continue using this SDK, or you can try to update with 'pip install cvat-sdk'.`

I guess there may be some version inconsistencies.

**Comment by zhiltsov-max** [comment]
> I guess there may be some version inconsistencies.

Yes, you should try to install the matching SDK version with `pip install 'cvat-sdk==2.3.0' 'cvat-cli==2.3.0'`.

**Comment by dmg-ai** [comment]
> > I guess there may be some version inconsistencies.
> 
> Yes, you should try to install the matching SDK version with `pip install 'cvat-sdk==2.3.0' 'cvat-cli==2.3.0'`.

Thanks for the answer! But it seems I’m talking about something a little different: the inconsistency between the UI CVAT Server version and the `cvat-sdk`, `cvat-cli` versions. And that's why updating libraries versions will not fix the problem.

<img width="779" alt="image" src="https://github.com/opencv/cvat/assets/55949105/835dd4e2-3bab-4805-ba96-819a3444bb07">

P.S: upgraded both `cvat-sdk` and `cvat-cli` libraries to `2.7.2` versions and it still doesn't work.

**Comment by zhiltsov-max** [comment]
If you're using SDK or CLI, you need their versions to match the server version. From the conversation above, your server has version 2.3, so your SDK and CLI should both be 2.3.0.

**Comment by A3itor** [comment]
Hello,

I am using cvat-sdk version v2.11 (more recent than v2.4.0) and I have found the same exact issue with the tasks low level API. See:

from cvat_sdk.api_client import Configuration, ApiClient
configuration = Configuration(host=host, username=username,  password=password)

with ApiClient(configuration) as api_client:
    tasks = api_client.tasks_api.list()[0].results
    for task in tasks:
        print(task.id)
           
Surprinsingly enough, api_client.tasks_api.list()[0] has 'count' = 12 (I have 10 tasks in page 1, and 2 in page 2)

How can 

> Hi, this problem was fixed in #5557 (v2.4.0+). Prior this change, only the first page (up to 10 entries) had been requested in this function. If you want to retrieve all the list, please use the low-level API and the `get_paginated_collection()` function:
> 
> ```python
> from cvat_sdk.core.helpers import get_paginated_collection
> from cvat_sdk.core.proxies.tasks import Task
> from cvat_sdk import make_client
> 
> format_name = "CVAT for images 1.1"
> 
> with make_client("localhost", port=8080, credentials=("admin", "pass")) as client:
>     tasks = [
>         Task(client, m)
>         for m in get_paginated_collection(
>             client.api_client.projects_api.list_tasks_endpoint, id=project_id
>         )
>     ]
> 
>     for task in tasks:
>         task.export_dataset(format_name, f"{task.id}-{task.name}-{format_name}.zip")
>         
> ```

Hello,

I am using cvat-sdk version v2.11 (more recent than v2.4.0) and I have found the same exact issue with the tasks low level API. See:

from cvat_sdk.api_client import Configuration, ApiClient
configuration = Configuration(host=host, username=username,  password=password)

with ApiClient(configuration) as api_client:
    tasks = api_client.tasks_api.list()[0].results
    for task in tasks:
        print(task.id)
 
 This will only print the ids of the first 10 tasks. Surprinsingly enough, api_client.tasks_api.list()[0] has 'count' = 12 (I have 10 tasks in page 1, and 2 in page 2)

Using the projects low level api proposed here raises an error at this line client.api_client.projects_api.list_tasks_endpoint, id=project_id saying there is no such method. Any help would be really appreciated.
